### DISCUSSION TITLE
UnknownHostException when using VPN

### DISCUSSION THREAD

**Discussion by Badzi007** [eclipse-milo/milo#839] [General]
Hello everyone,
i create this issue to get some help about using VPN to connect into Prosys OPC UA simulator server putted into a laptop.
when i use VPN to connect to this simulator server with another laptop (im using ip adress into endpoint like this opc.tcp://10.130.1.13:4840/OPCUA/SimulationServer ) .
the library connect to laptop and get his name  LAPTOP-XXXXXXX.home
after that when try to connect i get this exception:  java.net.UnknownHostException: LAPTOP-XXXXXXX.home: nodename nor servname provided, or not known

but it work fine without VPN when im connected directly to the local lan with the same laptop.

also i try a OPC UA client dev with python work fine (FreeOpcUa)

Could you please help me to find why i have this problem i check all what i know also in debug mod to know whats wrong but im stuck.

thanks in advance for your help.
im eclipse milo 0.6.0  with JDK zulu 8
You can see bellow screen of the stack trace and logs also wireshark trace:

![image](https://user-images.githubusercontent.com/55756504/117786673-0e29b000-b246-11eb-8942-83b1dd01315d.png)
![image](https://user-images.githubusercontent.com/55756504/117780037-8771d480-b23f-11eb-9d47-a4c053f8f097.png)
![image](https://user-images.githubusercontent.com/55756504/117785102-8b542580-b244-11eb-84e2-5e11aa3c4b79.png)

**Comment by kevinherron** [comment]
I think the server is returning an EndpointDescription with a hostname that contains the host `LAPTOP-MMHLFQGB.home` rather than the IP address you used to connect. 

You'll either have to add it to your hosts file so that your computer can resolve it or do something like mentioned in [this Stack Overflow post](https://stackoverflow.com/a/63345707/85927).
